hermes-common/src/main/java/pl/allegro/tech/hermes/infrastructure/logback/AggregatingTurboFilter.java

```diff
@@ -110,6 +119,15 @@ private boolean isAggregatedLog(Marker marker) {
         return marker != null && (marker.equals(MARKER) || marker.contains(MARKER));
     }
 
+    private Optional<Throwable> extractLastParamThrowable(Object[] params) {
+        return Optional.ofNullable(params)
+                .map(Arrays::stream)
+                .flatMap(a -> a.skip(params.length - 1)
```

Comment:
Is interested `Throwable` always on a last place?

Reply:
When calling `logger.error("some message {}", "something", exception);` - yes, it will be included as a standard param, this is why I'm fixing it (I'm taking it out, and storing it as the last exception thrown, and removing it from the passed params, so it won't be considered as a part of the "aggregating key").
When calling `logger.error("some message", exception);` - no, it won't be, it will be passed as the `throwable` argument.

So it depends on the exact method API, and slf4j has multiple variants of each.

Reply:
btw similar logic exists in logback, see `ch.qos.logback.classic.spi.LoggingEvent` class, some snippet when creating `LoggingEvent` instance:

```java
    this.argumentArray = argArray;

    if(throwable == null) {
      throwable = extractThrowableAnRearrangeArguments(argArray); // <---- here
    }

    if (throwable != null) {
      this.throwableProxy = new ThrowableProxy(throwable);
      LoggerContext lc = logger.getLoggerContext();
      if (lc.isPackagingDataEnabled()) {
        this.throwableProxy.calculatePackagingData();
      }
    }
```

...but it all takes place later (after TurboFilters are called), so I have to take care of it by myself and remove the exception.

Reply:
Oh, and the difference is I pass the exception as the `throwable` argument, not as a standard param. But in the end it doesn't matter.

Reply:
ok, got it :)